addons/certificate/models/certificate.py

```diff
@@ -0,0 +1,227 @@
+            # finally PEM. If none succeeded, we raise an error.
+            try:
+                cert = x509.load_der_x509_certificate(content)
+                certificate.content_format = 'der'
+            except ValueError:
+                pass
+            try:
+                pkcs12_password = certificate.pkcs12_password.encode('utf-8') if certificate.pkcs12_password else None
+                key, cert, _additionnal_certs = pkcs12.load_key_and_certificates(content, pkcs12_password)
+                certificate.content_format = 'pkcs12'
+            except ValueError:
+                pass
+            try:
+                cert = x509.load_pem_x509_certificate(content)
+                certificate.content_format = 'pem'
+            except ValueError:
+                pass
+
+            if not cert:
+                certificate.content = None
+                raise UserError(_("The certificate could not be loaded."))
+
+            # Extract certificate data
+            certificate.pem_certificate = base64.b64encode(cert.public_bytes(Encoding.PEM))
+            certificate.date_start = cert.not_valid_before
+            certificate.date_end = cert.not_valid_after
+            certificate.serial_number = cert.serial_number
+
+            # Create the private key in case of PKCS12 File
+            if key:
+                certificate.private_key = self.env['certificate.key'].create({
+                    'name': (certificate.name or "") + ".key",
+                    'content': base64.b64encode(key.private_bytes(
+                        encoding=Encoding.PEM,
+                        format=serialization.PrivateFormat.PKCS8,
+                        encryption_algorithm=serialization.NoEncryption()
+                    )),
+                    'company_id': certificate.company_id.id,
+                })
+
+    @api.depends('date_start', 'date_end')
+    def _compute_is_valid(self):
+        # Certificate dates are UTC timezoned
+        # https://cryptography.io/en/latest/x509/reference/#cryptography.x509.Certificate.not_valid_after
+        utc_now = datetime.datetime.now(datetime.timezone.utc)
+        for certificate in self:
+            date_start = certificate.date_start.replace(tzinfo=datetime.timezone.utc)
+            date_end = certificate.date_end.replace(tzinfo=datetime.timezone.utc)
+            certificate.is_valid = not (utc_now < date_start or utc_now > date_end)
+
+    def _search_is_valid(self, operator, value):
+        if operator not in ['=', '!='] or not isinstance(value, bool):
+            raise UserError(_('Operation not supported'))
+        if operator != '=':
+            value = not value
+        certificates = self.env['certificate.certificate'].search([('active', '=', True), ('company_id', '=', self.env.company.id)]) \
+                                                          .filtered(lambda cert: cert.is_valid == value)
```

Comment:
Unfortunately, it is not possible to do it without the filter since this function actually defines how `search` should behave when searching on `is_valid`. Searching on `is_valid` its definition leads to infinite recursion.